homeassistant/components/devolo_home_network/__init__.py

```diff
@@ -53,6 +54,14 @@ async def async_update_connected_plc_devices() -> dict[str, Any]:
         except DeviceUnavailable as err:
             raise UpdateFailed(err) from err
 
+    async def async_update_firmware_available() -> dict[str, Any]:
+        """Fetch data from API endpoint."""
+        try:
+            async with async_timeout.timeout(10):
+                return await device.device.async_check_firmware_available()  # type: ignore[no-any-return, union-attr]
```

Comment:
It would be nice to get this typed in the upstream lib.

https://www.python.org/dev/peps/pep-0561/

Reply:
It is typed, but with dict[str, Any] as its the JSON the device returns to me.